geomstats/geometry/special_euclidean.py

```diff
@@ -727,7 +727,449 @@ def _exponential_matrix(self, rot_vec):
         return exponential_mat
 
 
-class SpecialEuclidean(_SpecialEuclidean3Vectors, _SpecialEuclideanMatrices):
+class _SpecialEuclidean2Vectors(LieGroup):
+    """Class for the special euclidean group in 2d, SE(2).
+
+    i.e. the Lie group of rigid transformations. Elements of SE(32 can either
+    be represented as vectors (in 2d) or as matrices in general. The matrix
+    representation corresponds to homogeneous coordinates. This class is
+    specific to the vector representation of rotations. For the matrix
+    representation use the SpecialEuclidean class and set `n=2`.
+
+    Parameter
+    ---------
+    epsilon : float
+        Precision to use for calculations involving potential
+        division by 0 in rotations.
+        Optional, default: 0.
+    """
+
+    def __init__(self, epsilon=0.):
+        super(_SpecialEuclidean2Vectors, self).__init__(
+            dim=3, default_point_type='vector')
+
+        self.n = 2
+        self.epsilon = epsilon
+        self.rotations = SpecialOrthogonal(
+            n=2, point_type='vector', epsilon=epsilon)
+        self.translations = Euclidean(dim=2)
+
+    def get_identity(self, point_type=None):
+        """Get the identity of the group.
+
+        Parameters
+        ----------
+        point_type : str, {'vector', 'matrix'}
+            The point_type of the returned value.
+            Optional, default: self.default_point_type
+
+        Returns
+        -------
+        identity : array-like, shape={[dim], [n + 1, n + 1]}
+        """
+        if point_type is None:
+            point_type = self.default_point_type
+        identity = gs.zeros(self.dim)
+        if point_type == 'matrix':
```

Comment:
I think these two lines can be removed, as SpecialEuclidean..Vector only considers point_type vector